pyk/src/pyk/kore/rpc.py

```diff
@@ -56,6 +56,42 @@ def __init__(self, message: str, code: int, data: Any = None):
 
 
 class Transport(ContextManager['Transport'], ABC):
+    _bug_report: BugReport | None
+    _bug_report_id: str | None
+
+    def __init__(self, bug_report_id: str | None = None, bug_report: BugReport | None = None) -> None:
+        self._bug_report_id = bug_report_id
+        self._bug_report = bug_report
+
+    def request_with_bug_report(self, req: str, request_id: int) -> str:
+        base_name = self._bug_report_id if self._bug_report_id is not None else 'kore_rpc'
+        req_name = f'{base_name}/{self._client_id}/{request_id:03}'
+        if self._bug_report:
+            bug_report_request = f'{req_name}_request.json'
+            self._bug_report.add_file_contents(req, Path(bug_report_request))
+            self._bug_report.add_command(self.command(req_name, bug_report_request))
+
+        server_addr = self.description()
+        _LOGGER.debug(f'Sending request to {server_addr}: {req}')
+        resp = self.request(req)
+        if not resp:
+            raise RuntimeError('Empty response received')
+        _LOGGER.debug(f'Received response from {server_addr}: {resp}')
+
+        if self._bug_report:
+            bug_report_response = f'{req_name}_response.json'
+            self._bug_report.add_file_contents(resp, Path(bug_report_response))
+            self._bug_report.add_command(
+                [
+                    'diff',
+                    '-b',
+                    '-s',
+                    f'{req_name}_actual.json',
+                    f'{req_name}_response.json',
+                ]
+            )
+        return resp
+
     @abstractmethod
     def request(self, req: str) -> str: ...
```

Comment:
Can become private, `request_with_bug_report` then can be renamed to `request`. All the private abstract methods for `request` can be moved right after it.

Reply:
Fixed.